enclave/sgx/collateralinfo.c

```diff
@@ -169,6 +170,123 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
     return result;
 }
 
+void oe_prealloc_quote_verification_collateral_args(
+oe_get_sgx_quote_verification_collateral_args_t* buf)
+{
+    /*These numbers are just estimates of buffer sizes*/
+    size_t prealloc_tcb_info_size = 5000;
+    size_t prealloc_tcb_info_issuer_chain_size = 3000;
+    size_t prealloc_pck_crl_size = 600;
+    size_t prealloc_root_ca_crl_size = 600;
+    size_t prealloc_pck_crl_issuer_chain_size = 3000;
+    size_t prealloc_qe_identity_issuer_chain_size = 3000;
+    size_t prealloc_qe_identity_size = 700;
```

Comment:
If these values are too big, we waste memory; if too small, we double call. According to current PCK cache, actual sizes are 2125, 1893, 460, 448, 1909, 1893, and 498. Your pre-set values are reasonable. However, the sizes differ by time and hardware. It's hard to get them right. I suggest we make the size values static (no pre-set) and remember the retrieved sizes. It pretty much guarantees we only double call once per execution.